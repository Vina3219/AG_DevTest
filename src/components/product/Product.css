.product__container {
  display: flex;
  flex-direction: column;
  background-color: white;
  height: 400px;
  margin: 8px;
}

.product-image__container {
  height: 250px;

  display: flex;
  justify-content: center;
  align-items: center;
}
.product__image {
  height: 90%;
}

.product__detail {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  text-align: left;
  border-top: 1px solid #f5f5f5;
  padding: 24px;
}

.product__detail h3 {
  font-weight: 400;
  color: #8a8a8a;
  margin: 0;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2; /* number of lines to show */
  line-clamp: 2;
  -webkit-box-orient: vertical;
}

.product__detail .product__description {
  color: #bebebe;
  margin-top: 4px;
  margin-bottom: 4px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 2; /* number of lines to show */
  line-clamp: 2;
  -webkit-box-orient: vertical;
}

.product__detail .product__price {
  color: #202020;
  font-weight: 700;
}
