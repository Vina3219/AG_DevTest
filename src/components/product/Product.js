import React from "react";
import "./Product.css";

const Product = ({ product }) => {
  return (
    <div className="product__container">
      <div className="product-image__container">
        <img
          className="product__image"
          src={product.product_image}
          alt={product.product_name}
        ></img>
      </div>

      <div className="product__detail">
        <h3>{product.product_name}</h3>
        <span className="product__description">{product.description}</span>
        <span className="product__price">{product.price}</span>
      </div>
    </div>
  );
};

export default Product;
