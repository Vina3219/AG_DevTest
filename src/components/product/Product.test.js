import { render } from "@testing-library/react";
import Product from "./Product";

describe("Product", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Product
        product={{
          product_image: "test_image.url",
          product_name: "test_name",
          description: "test product description",
          price: "123",
        }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
