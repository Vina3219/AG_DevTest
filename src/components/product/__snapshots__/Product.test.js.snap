// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="product__container"
  >
    <div
      class="product-image__container"
    >
      <img
        alt="test_name"
        class="product__image"
        src="test_image.url"
      />
    </div>
    <div
      class="product__detail"
    >
      <h3>
        test_name
      </h3>
      <span
        class="product__description"
      >
        test product description
      </span>
      <span
        class="product__price"
      >
        123
      </span>
    </div>
  </div>
</DocumentFragment>
`;
